crates/backend/src/codegen.rs

```diff
@@ -146,131 +146,180 @@ impl ToTokens for ast::Struct {
         let name_str = self.js_name.to_string();
         let name_len = name_str.len() as u32;
         let name_chars = name_str.chars().map(|c| c as u32);
-        let new_fn = Ident::new(&shared::new_function(&name_str), Span::call_site());
-        let free_fn = Ident::new(&shared::free_function(&name_str), Span::call_site());
-        let free_fn_const = Ident::new(&format!("{}__const", free_fn), free_fn.span());
-        (quote! {
-            #[automatically_derived]
-            impl wasm_bindgen::describe::WasmDescribe for #name {
-                fn describe() {
-                    use wasm_bindgen::__wbindgen_if_not_std;
-                    __wbindgen_if_not_std! {
-                        compile_error! {
-                            "exporting a class to JS requires the `std` feature to \
-                             be enabled in the `wasm-bindgen` crate"
-                        }
+        if self.transparent {
+            let field = &self.fields[0];
+            let field_ty = &field.ty;
+            (quote! {
+                #[automatically_derived]
+                impl wasm_bindgen::describe::WasmDescribe for #name {
+                    fn describe() {
+                        #field_ty::describe();
                     }
-                    use wasm_bindgen::describe::*;
-                    inform(RUST_STRUCT);
-                    inform(#name_len);
-                    #(inform(#name_chars);)*
                 }
-            }
 
-            #[automatically_derived]
-            impl wasm_bindgen::convert::IntoWasmAbi for #name {
-                type Abi = u32;
+                #[automatically_derived]
+                impl wasm_bindgen::convert::IntoWasmAbi for #name {
+                    type Abi = u32;
 
-                fn into_abi(self) -> u32 {
-                    use wasm_bindgen::__rt::std::boxed::Box;
-                    use wasm_bindgen::__rt::WasmRefCell;
-                    Box::into_raw(Box::new(WasmRefCell::new(self))) as u32
+                    fn into_abi(self) -> u32 {
+                        // SAFETY: repr(transparent) means, by definition, that we can transmute
+                        let value: #field_ty = unsafe { std::mem::transmute(self) };
```

Comment:
Instead of using `transmute` can this construct an instance of the type directly? (same for accessing the internal value)

Reply:
That's what I did first, but this becomes complicated when the type has one or more zero-sized fields. I can't really think of a simple way to construct any type without bloating `ast::Struct`, but maybe I'm missing something?